Material_TimePicker/src/main/ets/components/TimePickerModel.ets

```diff
@@ -0,0 +1,267 @@
+    if (options.cancel != undefined) this.addOnCancelListener(options.cancel)
+    if (options.onPositiveButtonClick != undefined) this.addOnPositiveButtonClickListener(options.onPositiveButtonClick)
+    if (options.onNegativeButtonClick != undefined) this.addOnNegativeButtonClickListener(options.onNegativeButtonClick)
+    if (options.dismiss != undefined) this.addOnDismissListener(options.dismiss)
+    if (options.keyboardSymbol != undefined) this.setKeyboardSymbol(options.keyboardSymbol)
+    if (options.clockSymbol != undefined) this.setClockSymbol(options.clockSymbol)
+    if (options.clockFaceBackgroundColor != undefined) this.setClockFaceBackgroundColor(options.clockFaceBackgroundColor)
+    if (options.clockHandColor != undefined) this.setClockHandColor(options.clockHandColor)
+    if (options.clockNumberTextColor != undefined) this.setClockNumberTextColor(options.clockNumberTextColor)
+    if (options.clockInputTitle != undefined) this.clockInputTitle = options.clockInputTitle
+    if (options.textInputTitle != undefined) this.textInputTitle = options.textInputTitle
+    if (options.inputMethod != undefined) this.setInputMethod(options.inputMethod)
+    if (options.titleBackgroundColor != undefined) this.titleBackgroundColor = options.titleBackgroundColor
+    if (options.titleFontColor != undefined) this.titleFontColor = options.titleFontColor
+    if (options.labelColor != undefined) this.labelColor = options.labelColor
+    if (options.materialCircleRadius != undefined) this.materialCircleRadius = options.materialCircleRadius
+    if (options.selectorSize != undefined) this.pointerDiameter = options.selectorSize
+
+  }
+  onPositiveButtonClick(time:string) {
+    this.onPositiveButtonClickListeners.forEach(element=>{element(time)})
+  }
+  onNegativeButtonClick() {
+    this.onNegativeButtonClickListeners.forEach(element=>element())
+  }
+  onCancel() {
+    this.onCancelListeners.forEach(element=>element())
+  }
+  onDismiss() {
+    this.onDismissListeners.forEach(element=>element())
+  }
+  addOnPositiveButtonClickListener(listener:((string)=>void)[]) {
+    listener.forEach(element=>this.onPositiveButtonClickListeners=this.onPositiveButtonClickListeners.concat(element))
+  }
+
+  addOnNegativeButtonClickListener(listener:(()=>void)[]) {
+    listener.forEach(element=>this.onNegativeButtonClickListeners=this.onNegativeButtonClickListeners.concat(element))
+  }
+
+  addOnCancelListener(listener:(()=>void)[]) {
+    listener.forEach(element=>this.onCancelListeners=this.onCancelListeners.concat(element))
+  }
+
+  addOnDismissListener(listener:(()=>void)[]) {
+    listener.forEach(element=>this.onDismissListeners=this.onDismissListeners.concat(element))
+  }
+
+  setKeyboardSymbol(keyBoardSymbol: Resource) {
+    this.inputSymbol[0] = keyBoardSymbol
+  }
+
+  setClockSymbol(clockSymbol: Resource) {
+    this.inputSymbol[1] = clockSymbol
+  }
+
+  setInputMethod(method: InputMethod) {
+    if (this.inputMethod != method) {
```

Comment:
handle null and undefined validation in all setter methods

Reply:
done